tpm2/tpm2.go

```diff
@@ -1003,8 +1013,8 @@ func decodeSign(buf []byte) (*Signature, error) {
 
 // Sign computes a signature for digest using a given loaded key. Signature
 // algorithm depends on the key type.
-func Sign(rw io.ReadWriter, key tpmutil.Handle, password string, digest []byte) (*Signature, error) {
-	cmd, err := encodeSign(key, password, digest)
+func Sign(rw io.ReadWriter, key tpmutil.Handle, password string, digest []byte, sigScheme, hashAlg Algorithm, count uint32) (*Signature, error) {
```

Comment:
Ah, I totally forgot, `TPMT_SIG_SCHEME` is already (almost) implemented: https://github.com/google/go-tpm/blob/master/tpm2/structures.go#L256

Could you add `count` field (and handling of ECDAA) to it instead and use a `*SigScheme` argument for `Sign`?

Reply:
I agree. Using SigScheme in Sign sounds like the right choice for the API.